pkg/types/valid/namespacedname.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// pkg/api/validation includes pkg/types, so we need a new subpackage to
+// avoid circular imports.
+package valid
+
+import (
+	"errors"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/validation"
+)
+
+// NamespacedName comprises a resource name, with a mandatory namespace,
+// rendered as "<namespace>/<name>".  Being a type captures intent and
+// helps make sure that UIDs, namespaced names and non-namespaced names
+// do not get conflated in code.  It also provides a convenient place for
+// utility functions to do things like splitting, joining and rendering.
+type NamespacedName struct {
+	namespace string
+	name      string
+}
+
+// NewNamespacedName is the constructor, and checks that inputs are validly formatted
+func NewNamespacedName(namespace, name string) (NamespacedName, error) {
+	if valid, errMsg := validation.ValidateNamespaceName(namespace, false); !valid {
+		err := errors.New("namespace \"" + namespace + "\": " + errMsg)
+		return NamespacedName{}, err
+	}
+	if valid, errMsg := validation.ValidateServiceName(namespace, false); !valid {
```

Comment:
You are right.  I'm still stewing over whether to simply not do validation, or expand it to handle all cases.  As the code stands now, it only works correctly for services, and other entities that follow the same convention as services.  This is bad.